base/server/src/main/java/com/netscape/cms/servlet/request/CMSRequestDAO.java

```diff
@@ -63,6 +65,65 @@ public abstract class CMSRequestDAO {
     public CMSRequestDAO() {
     }
 
+    /**
+     * Finds list of requests matching the specified search filter.
+     *
+     * If the filter corresponds to a VLV search, then that search is executed and the pageSize
+     * and start parameters are used. Otherwise, the maxResults and maxTime parameters are
+     * used in the regularly indexed search.
+     *
+     * @param filter - ldap search filter
+     * @param start - start position
+     * @param pageSize - page size for search
+     * @param maxTime - max time for normal search
+     * @param uriInfo - uri context of request
+     * @return collection of key request info
+     * @throws EBaseException
+     */
+    public CMSRequestInfos listCMSRequests(String filter, RequestId start, int pageSize, int maxTime,
+            UriInfo uriInfo) throws EBaseException {
+
+    logger.info("CMSRequestDAO: Searching for requests with filter " + filter);
+
+    CMSRequestInfos ret = new CMSRequestInfos();
+
+        logger.debug("CMSRequestDAO: performing paged search");
+
+        RecordPagedList<RequestRecord> pagedList = requestRepository.getPagedRequestsByFilter(
+                start,
+                filter,
+                pageSize + 1,
+                "requestId");
+
+
+        Iterator<RequestRecord> reqs = pagedList.getCurrentPageEntries();
+        while(reqs.hasNext()) {
+            Request request = reqs.next().toRequest();
+            logger.debug("- " + request.getRequestId().toHexString());
+            ret.addEntry(createCMSRequestInfo(request, uriInfo));
+        }
+        ret.setTotal(requestRepository.getTotlaRequestsByFilter(filter));
```

Comment:
IIUC this operation will execute the same search again to get the total. Maybe for future improvement, is it possible for `RecordPagedList` to count the total so we don't have to do an additional search?